### ISSUE TITLE
overriding a resolver in config_dev.yml is ignored

### ISSUE THREAD

**Issue by Laith** [liip/LiipImagineBundle#611]
By design config_prod.yml and config_dev.yml should override the configuration in config.yml. though the following will fail 

config.yml
```
liip_imagine:
    loaders:
        .....
    resolvers:
        images:
            web_path:
                web_root: %kernel.root_dir%/../web/media
                cache_prefix: cache
```

config_dev.yml
```
liip_imagine:
    resolvers:
        images:
            aws_s3: 
                client_config:
                    key: %amazon_s3_key%
                    secret: %amazon_s3_secret%
                    region: %amazon_s3_region%
                bucket:     %amazon_s3_bucket%
                cache_prefix: cache
```
the issue seems to be here https://github.com/liip/LiipImagineBundle/blob/master/DependencyInjection/LiipImagineExtension.php#L98

since `key` will get the first key of the array while the behavior should be the last key of the array

I guess loaders will face the same issue since its loaded the same way, didn't test it yet

thoughts?

**Comment by dincho** [comment]
As workaround you may configure all your resolvers in config.yml with different names:
```
liip_imagine:
    loaders:
        .....
    resolvers:
        images_local:
            web_path:
                web_root: %kernel.root_dir%/../web/media
                cache_prefix: cache        images:
        images_s3:
            aws_s3: 
                client_config:
                    key: %amazon_s3_key%
                    secret: %amazon_s3_secret%
                    region: %amazon_s3_region%
                bucket:     %amazon_s3_bucket%
                cache_prefix: cache
```

 then in config_prod.yml just set the correct one:

```
liip_imagine:
    cache:      images_s3
```

**Comment by makasim** [comment]
Fixed in 1.4.0
